src/generators/default_map_generator_job.cpp

```diff
@@ -1213,67 +1186,79 @@ std::string default_map_generator_job::default_generate_map(size_t width, size_t
-					const std::map<location, std::string>::const_iterator forest_name = forest_names.find(loc);
-					if(forest_name == forest_names.end()) {
-						for(size_t ntry = 0; ntry != 30 && (ntry == 0 || used_names.count(name) > 0); ++ntry) {
-							base_name = base_name_generator->generate();
-							name = forest_name_generator->generate({{"base",  base_name}});
-						}
-						forest_names.insert(std::pair<location, std::string>(loc, base_name));
-						// name all connected forest tiles accordingly
-						flood_name(loc, base_name, forest_names, t_translation::ALL_FORESTS, terrain, width, height, 0, misc_labels, name);
+			} else if(t_translation::terrain_matches(terr, t_translation::ALL_FORESTS)) {
+				// If the forest tile is not named yet, name it
+				const std::map<location, std::string>::const_iterator forest_name = forest_names.find(loc);
+				if(forest_name == forest_names.end()) {
+					for(size_t ntry = 0; ntry != 30 && (ntry == 0 || used_names.count(name) > 0); ++ntry) {
+						base_name = base_name_generator->generate();
+						name = forest_name_generator->generate({{"base",  base_name}});
 					}
+					forest_names.insert(std::pair<location, std::string>(loc, base_name));
+					// name all connected forest tiles accordingly
+					flood_name(loc, base_name, forest_names, t_translation::ALL_FORESTS, terrain, width, height, 0, misc_labels, name);
 				}
-				else if(t_translation::terrain_matches(terr, t_translation::ALL_SWAMPS)) {
-					//if the swamp tile is not named yet, name it
-					const std::map<location, std::string>::const_iterator swamp_name = swamp_names.find(loc);
-					if(swamp_name == swamp_names.end()) {
-						for(size_t ntry = 0; ntry != 30 && (ntry == 0 || used_names.count(name) > 0); ++ntry) {
-							base_name = base_name_generator->generate();
-							name = swamp_name_generator->generate({{"base",  base_name}});
-						}
-						swamp_names.insert(std::pair<location, std::string>(loc, base_name));
-						// name all connected swamp tiles accordingly
-						flood_name(loc, base_name, swamp_names, t_translation::ALL_SWAMPS, terrain, width, height, 0, misc_labels, name);
+			} else if(t_translation::terrain_matches(terr, t_translation::ALL_SWAMPS)) {
+				// If the swamp tile is not named yet, name it
+				const std::map<location, std::string>::const_iterator swamp_name = swamp_names.find(loc);
+				if(swamp_name == swamp_names.end()) {
+					for(size_t ntry = 0; ntry != 30 && (ntry == 0 || used_names.count(name) > 0); ++ntry) {
+						base_name = base_name_generator->generate();
+						name = swamp_name_generator->generate({{"base",  base_name}});
 					}
+					swamp_names.insert(std::pair<location, std::string>(loc, base_name));
+					// name all connected swamp tiles accordingly
+					flood_name(loc, base_name, swamp_names, t_translation::ALL_SWAMPS, terrain, width, height, 0, misc_labels, name);
 				}
-			}//for(y)
-		}//for(x)
-	}//if(misc_labels)
+			}
+		}
+	}
+
+	place_villages:
```

Comment:
_If_ you're going to use labels at all (which I don't recommend in this situation), they should typically be indented one less than the surrounding code.